src/custom/utils/paraswap.ts

```diff
@@ -0,0 +1,86 @@
+import { OrderKind } from '@gnosis.pm/gp-v2-contracts'
+import { ParaSwap, SwapSide, NetworkID } from 'paraswap'
+import { toErc20Address } from 'utils/tokens'
+import { PriceQuoteParams } from 'utils/operator'
+import { OptimalRatesWithPartnerFees, RateOptions } from 'paraswap/build/types'
+import { ChainId } from '@uniswap/sdk'
+import { PriceInformation } from 'state/price/reducer'
+import { getTokensFromMarket } from './misc'
+
+type ParaSwapPriceQuote = OptimalRatesWithPartnerFees
+
+// Provided manually just to make sure it matches what GPv2 backend is using, although the value used  is the current SDK default
+const API_URL = 'https://apiv4.paraswap.io/v2'
+
+const parSwapLibs: Map<ChainId, ParaSwap> = new Map()
+
+function getParaswapChainId(chainId: ChainId): NetworkID | null {
+  switch (chainId) {
+    // case ChainId.RINKEBY: // Rinkeby is supported by the SDK but not from the API
+    // case ChainId.XDAI: // xDAI not supported for now
+    case ChainId.MAINNET:
+    case ChainId.KOVAN:
+    case ChainId.ROPSTEN:
+      return chainId
+
+    default:
+      // Unsupported network
+      return null
+  }
+}
+
+export function toPriceInformation(priceRaw: ParaSwapPriceQuote | null): PriceInformation | null {
+  if (!priceRaw || !priceRaw.details) {
+    return null
+  }
+
```

Comment:
Is `side` what paraswap returns as a prop or is this your own name? Better to keep the name consistent as "kind" as we do elsewhere

Reply:
side is their name, I can rename it when extracting the prop, but I think is simple enough like this. WDYT?

Reply:
Nah all good